framework/src/modules/chain/chain.js

```diff
@@ -558,6 +558,14 @@ module.exports = class Chain {
 				);
 			}
 			this.channel.publish('chain:blocks:change', block);
+
+			// If block version is 2
```

Comment:
the comment mentions block version but the if checks for `loader.syncing()` can this comment be improved with more details?

Reply:
Actually, it can be removed. Good catch!

Reply:
Related issue: #3875

Reply:
Agree, it feels not related to this line of code like this.